Microsoft.Toolkit.Uwp.UI.Controls.Layout/GridSplitter/SplitBase.cs

```diff
@@ -0,0 +1,220 @@
+            DependencyProperty.Register(nameof(ResizeDirection), typeof(ContentResizeDirection), typeof(SplitBase), new PropertyMetadata(ContentResizeDirection.Vertical));
+
+        /// <summary>
+        /// Gets or sets the control that the <see cref="SplitBase"/> is resizing. Be default, this will be the visual ancestor of the <see cref="SplitBase"/>.
+        /// </summary>
+        public FrameworkElement TargetControl
+        {
+            get { return (FrameworkElement)GetValue(TargetControlProperty); }
+            set { SetValue(TargetControlProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="TargetControl"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty TargetControlProperty =
+            DependencyProperty.Register(nameof(TargetControl), typeof(FrameworkElement), typeof(SplitBase), new PropertyMetadata(null, OnTargetControlChanged));
+
+        private static void OnTargetControlChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            // Check if our width can be manipulated
+            if (d is SplitBase splitterBase && e.NewValue is FrameworkElement element)
+            {
+                // TODO: For Auto we might want to do detection logic (TBD) here first?
+                if (splitterBase.ResizeDirection != ContentResizeDirection.Horizontal && double.IsNaN(element.Width))
+                {
+                    element.Width = element.DesiredSize.Width;
+                }
+
+                if (splitterBase.ResizeDirection != ContentResizeDirection.Vertical && double.IsNaN(element.Height))
+                {
+                    element.Height = element.DesiredSize.Height;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the <see cref="SplitBase"/> control is resizing in the opposite direction.
+        /// </summary>
+        public bool InvertDragDirection
+        {
+            get { return (bool)GetValue(InvertDragDirectionProperty); }
+            set { SetValue(InvertDragDirectionProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="InvertDragDirection"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty InvertDragDirectionProperty =
+            DependencyProperty.Register(nameof(InvertDragDirection), typeof(bool), typeof(SplitBase), new PropertyMetadata(false));
+
+        /// <summary>
+        /// Check for new requested vertical size is valid or not
+        /// </summary>
+        /// <param name="target">Target control being resized</param>
+        /// <param name="verticalChange">The requested vertical change</param>
+        /// <returns>Bool result if requested vertical change is valid or not</returns>
+        protected bool IsValidHeight(FrameworkElement target, double verticalChange)
```

Comment:
These two functions are only used in ContentSizer, so we should put them in that class somewhere, eh?